src/main/webapp/resources/js/pages/projects/samples/components/MergeModal.jsx

```diff
@@ -0,0 +1,182 @@
+      form.setFieldsValue({
+        newName: "",
+      });
+    }
+  }, [form, renameSample]);
+
+  // Server validate new name
+  const validateName = async (name) => {
+    if (renameSample) {
+      return serverValidateSampleName(name);
+    } else {
+      return Promise.resolve();
+    }
+  };
+
+  const onSubmit = async () => {
+    let values;
+
+    try {
+      values = await form.validateFields();
+    } catch {
+      return;
+    }
+    const ids = samples.valid
+      .map((sample) => sample.id)
+      .filter((id) => id !== values.primary);
+
+    const { message } = await merge({
+      projectId,
+      request: {
+        ...values,
+        ids,
+      },
+    }).unwrap();
+
+    notification.success({
+      message: i18n("MergeModal.success"),
+      description: message,
+    });
+    onComplete();
+  };
+
+  // TODO: Handle rendering many samples?
+  // TODO: Handle rendering many locked samples?k
+  return (
+    <Modal
+      title={i18n("MergeModal.title")}
+      className="t-merge-modal"
+      visible={visible}
+      onOk={onSubmit}
+      okText={i18n("MergeModal.okText")}
+      okButtonProps={{
+        loading: isLoading,
+      }}
+      onCancel={onCancel}
+      cancelText={i18n("MergeModal.cancelText")}
+      width={600}
+    >
+      <Row gutter={[16, 16]}>
+        {samples.valid.length >= 2 ? (
+          <>
+            <Col span={24}>
+              <Alert
+                type="warning"
+                showIcon
+                message={i18n("MergeModal.metadata-warning")}
+              />
+            </Col>
+            {error !== undefined ? (
+              <Col span={24}>
+                <Alert
+                  showIcon
+                  type="error"
+                  message={error}
+                  closable
+                  onClose={() => setError(undefined)}
+                />
+              </Col>
+            ) : null}
+            <Col span={24}>
+              <Form form={form} layout="vertical" initialValues={initialValues}>
+                <Form.Item
+                  name="primary"
+                  label={i18n("MergeModal.input-primary")}
+                  required
+                >
+                  <Radio.Group>
+                    <Space direction="vertical">
+                      {samples.valid.map((sample) => {
+                        return (
+                          <Radio value={sample.id} key={`sample-${sample.id}`}>
+                            {sample.sampleName}
+                          </Radio>
+                        );
+                      })}
+                    </Space>
+                  </Radio.Group>
+                </Form.Item>
+                <Form.Item noStyle>
+                  <Checkbox
+                    className="t-custom-checkbox"
+                    checked={renameSample}
+                    onChange={(e) => setRenameSample(e.target.checked)}
+                  >
+                    Rename Sample
```

Comment:
Internationalize

Reply:
Updated in f183bbc1d3423715d5e1df07b89a9d8b5c0da8f7